Zend/zend_alloc.c

```diff
@@ -602,6 +613,28 @@ static void *zend_mm_mmap(size_t size)
 #endif
 }
 
+static void zend_mm_mprotect_ro(void* addr, size_t len)
+{
+#ifdef HAVE_MPROTECT
+	ZEND_MM_CHECK(mprotect(addr, len, PROT_READ) == 0, "Unable to mprotect memory as read-only");
```

Comment:
:+1: 

Noticed that we check for `mprotect()` only in `ext/opcache/config.m4`, so `HAVE_MPROTECT` is not defined when opcache is not built.

We should add `mprotect \` [here](https://github.com/php/php-src/blob/ab9a0295c566f9c34e395ffead8d030e60524cce/configure.ac#L626) to ensure it's always checked.

Maybe we can be defensive against autoconf issues here, since it's a security feature, and assume `mprotect()` always exists unless `_WIN32` is defined.